### ISSUE TITLE
Feature Request: Offer to open changelog when an extension is updated

### ISSUE THREAD

**Issue by ramya-rao-a** [Microsoft/vscode#21583]
With every update to VS Code, users now get a prompt to read the release notes.
It would be nice to have the same feature for extensions. Prompt to read the Changelog (if exists) on every update to the extension

**Comment by ramya-rao-a** [comment]
Slack has this "gift" icon. On hover it says "What's new". Clicking opens a side pane with the new features.

In the extensions pane, we now have the "Update to .." button when there is an update available.
We can use the same space for a "What's new" button when there is no update and changelog exists for the extension. Clicking on which should open the Changelog tab.

@stevencl Thoughts?
cc @Microsoft/vscode

**Comment by chrmarti** [comment]
I like the idea of increasing the visibility of extensions' progress. One issue might be that all extensions end up having that changelog button because (almost) all have a prior version. That would decrease its value as a 'highlight'.

**Comment by bpasero** [comment]
-1, I have set my extensions to auto update and I would never want to see any release editor opening. I am quite happy with this kind of silent updates.

**Comment by joaomoreno** [comment]
As long as it doesn't come as notification, I'm fine with that... The more I call `showInformationMessage`, the less chances I have to get into Heaven.

**Comment by stevencl** [comment]
A 'What's new' button after the user has updated an extension would be a decent solution. I definitely don't think we should be notifying the user.

Once the user closes VS Code and restarts, we would remove the What's New button. We could consider then exposing a 'View changelog' action in the gear/settings menu to provide access to the changelog if it exists. That way we could avoid the problem that @chrmarti points out.

**Comment by ramya-rao-a** [comment]
Agreed on the "no more notifications please" thought.

**Comment by eamodio** [comment]
I would LOVE to see a feature like this. I've posted some of my comments here: https://github.com/eamodio/vscode-gitlens/issues/222#issuecomment-354311075

Personally I think it is important for an extension to be able to provide details around an update (changelog, release notes, etc). But the current notification mechanism is far from ideal. IMO, it should be something discoverable but passive. 

I kind of like the idea of a badge on the extensions button (which would go away if you open the extension pane) if there have been any updates. And then individual badge each extension that has been updated -- that would bring you to a changelog or something like that. Although the downside, is lots of badges that might not get looked at and therefore not cleared, which would just become visual noise.

Another option, would still have the same badge on the extensions button, but the extensions pane would also have that badge as a button somewhere, and clicking that would open up a single screen, with a list of any updates. This could either be automatic (i.e. vscode will track the updates), manual (a new api for an extension to add an entry), or a combination of both. I think a combination would be ideal. Where vscode would track the simple, `Extension Foo was upgraded from 1.0 -> 2.0` (maybe with a changelog link if there is one), and an extension could add additional details (bounded) through a new api -- maybe something like `showInformationMessage` (`addUpdateMessage`)?

Thoughts?

**Comment by ramya-rao-a** [comment]
@bpasero This is something we should include in the notification improvement story?

**Comment by bpasero** [comment]
@ramya-rao-a yeah we briefly touched on this. what if the update notification for an extension would show within the extensions viewlet instead of the global notification area?

**Comment by ramya-rao-a** [comment]
That would work, are we planning to cover that as part of #22388?

**Comment by bpasero** [comment]
@ramya-rao-a no, we would probably need a new story for that one. What we have planned is to show the source of a notification more prominently so distinguish core notifications from extension notifications.

**Comment by moshfeu** [comment]
Is this implemented yet? Because I think I saw a notification in vscode about an extension been updated and button which opens the extension's changelog

**Comment by sandy081** [comment]
Not yet, but I will be starting to explore based on the feedback that is provided here. So ideas are very much welcome.

**Comment by sandy081** [comment]
There's discussion happening here - https://github.com/microsoft/vscode/issues/102139

**Comment by anburocky3** [comment]
Out of curiosity, I see some extensions found a way to accomplish this way. But is there any vscode documentation to implement it ourself?
